// Copyright 2024 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


message ComputeAlphaNetworkFirewallPolicy {
  string location = 1;
  string creation_timestamp = 2;
  string name = 3;
  string id = 4;
  string description = 5;
  string fingerprint = 6;
  string self_link = 7;
  string self_link_with_id = 8;
  int64 rule_tuple_count = 9;
  string region = 10;
  string project = 11;
}

message ApplyComputeAlphaNetworkFirewallPolicyRequest {
  ComputeAlphaNetworkFirewallPolicy resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteComputeAlphaNetworkFirewallPolicyRequest {
  string service_account_file = 1;
  ComputeAlphaNetworkFirewallPolicy resource = 2;
}

message ListComputeAlphaNetworkFirewallPolicyRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListComputeAlphaNetworkFirewallPolicyResponse {
  repeated ComputeAlphaNetworkFirewallPolicy items = 1;
}

service ComputeAlphaNetworkFirewallPolicyService {
  rpc ApplyComputeAlphaNetworkFirewallPolicy(ApplyComputeAlphaNetworkFirewallPolicyRequest) returns (ComputeAlphaNetworkFirewallPolicy);
  rpc DeleteComputeAlphaNetworkFirewallPolicy(DeleteComputeAlphaNetworkFirewallPolicyRequest) returns (google.protobuf.Empty);
  rpc ListComputeAlphaNetworkFirewallPolicy(ListComputeAlphaNetworkFirewallPolicyRequest) returns (ListComputeAlphaNetworkFirewallPolicyResponse);
}
